batchflow/tests/model_save_load_test.py

```diff
@@ -0,0 +1,177 @@
+        Same dataset with no shuffling should be used, so that on same iterations
+        the pipelines get same data
+        """
+        dataset, model_config = model_setup_images_clf(image_shape=(100, 100, 2))
+        config = {'model_class': TF_VGG7, 'model_config': model_config}
+
+        save_pipeline = (Pipeline()
+                         .init_variable('predictions', init_on_each_run=list)
+                         .init_model(mode, C('model_class'), 'model', C('model_config'))
+                         .to_array()
+                         .predict_model('model',
+                                        fetches='predictions', save_to=V('predictions', mode='a'),
+                                        images=B('images')))
+        load_pipeline = (Pipeline()
+                         .init_variable('predictions', init_on_each_run=list)
+                         .to_array()
+                         .predict_model('model',
+                                        fetches='predictions', save_to=V('predictions', mode='a'),
+                                        images=B('images')))
+
+        save_pipeline = (save_pipeline << dataset) << config
+        load_pipeline = (load_pipeline << dataset) << config
+        return save_pipeline, load_pipeline
+
+    @pytest.fixture
+    def torch_pipelines(self, model_setup_images_clf, mode):
+        """
+        Create pipelines with Torch models to be saved and to be loaded
+        see :meth:`~.TestModelSaveLoad.tf_pipelines` for details
+        """
+        dataset, model_config = model_setup_images_clf(image_shape=(2, 100, 100))
+        config = {'model_class': TORCH_VGG7, 'model_config': model_config}
+
+        save_pipeline = (Pipeline()
+                         .init_variable('predictions', init_on_each_run=list)
+                         .init_model(mode, C('model_class'), 'model', C('model_config'))
+                         .to_array(channels='first', dtype='float32')
+                         .predict_model('model', B('images'),
+                                        fetches='predictions', save_to=V('predictions', mode='a')))
+        load_pipeline = (Pipeline()
+                         .init_variable('predictions', init_on_each_run=list)
+                         .to_array(channels='first', dtype='float32')
+                         .predict_model('model', B('images'),
+                                        fetches='predictions', save_to=V('predictions', mode='a')))
+
+        save_pipeline = (save_pipeline << dataset) << config
+        load_pipeline = (load_pipeline << dataset) << config
+        return save_pipeline, load_pipeline
+
+    @pytest.fixture
+    def fixture_dict(self, tf_pipelines, torch_pipelines):
+        """
+        workaround for passing parametrized fixtures as parameters
+        """
+        return dict(tf_pipelines=tf_pipelines, torch_pipelines=torch_pipelines)
+
+    @pytest.mark.parametrize('fixture_name, model_args, model_kwargs',
```

Comment:
`model_args` and `model_kwargs` are used to take into account different order of parameters for `tf`/`torch` versions of `train`. Why can't you use the same logic for `predict`? That would make `tf_pipeline` and `torch_pipeline` look very similar. The only real difference would be in `to_array` call, and you can easily avoid that with pipeline config

Reply:
I thought about it, but we have to parametrize predict and train separately because in torch we have different sets of positional args in train and in predict, so I thought it would be over complicated. 
I'll try to make all parametrization in config and compare with current state